ui/src/components/Button/index.tsx

```diff
@@ -3,60 +3,62 @@ import React from 'react';
 import * as Components from '@/components';
 
 type CommonProps = {
-    id?: string;
-    title: string;
-    endIcon?: React.ReactElement;
-    startIcon?: React.ReactElement;
-    onClick?: (e: React.MouseEvent) => void;
-    disabled?: boolean;
-    className?: string;
-    childClassName?: string;
-    textSize?: string;
-    padding?: string;
-    children?: React.ReactNode;
     accordionConfig?: {
         contentElementId: string;
         expanded: Boolean;
     };
+    childClassName?: string;
+    children?: React.ReactNode;
+    className?: string;
+    disabled?: boolean;
+    endIcon?: React.ReactElement;
+    id?: string;
+    onClick?: (e: React.MouseEvent) => void;
+    padding?: string;
+    startIcon?: React.ReactElement;
+    textSize?: string;
+    title: string;
+    variant?: 'underlined' | 'block' | 'block-alt';
 };
 
 type ConditionalProps = { href: string; openNew?: boolean } | { href?: never; openNew?: never };
 
 type Props = CommonProps & ConditionalProps;
 
+const blockClasses =
+    'border-2 px-2.5 text-white-50 shadow-sm focus:ring-offset-2 children:border-0 children:text-white-50 ';
+
 const Button: React.FC<Props> = (props): React.ReactElement | null => {
     const parentStyles = React.useMemo(() => {
-        return `
-            group
-            inline-flex
-            items-center
-            outline-0
-            focus:ring-2
-            focus:ring-yellow-400
-            disabled:select-none
-            disabled:opacity-50
-            disabled:hover:cursor-not-allowed
-            ${props.className ?? ''}
-            `;
-    }, [props.className]);
+        let classes =
+            'disabled:select-none disabled:opacity-50 disabled:hover:cursor-not-allowed inline-flex items-center w-fit ';
+        switch (props.variant) {
```

Comment:
Is this to make tailwind work properly? I was going to suggest maybe doing an interpolated string, but I recall tailwind had issues with that

Reply:
Something to do with having to be a static string at build time

Reply:
String interpolation works; this is just the way I found that I felt kept the formatting nice and the conditions clear. Seems like I can do it with interpolation like this, if this seems better?

```
const classes = `disabled:select-none disabled:opacity-50 disabled:hover:cursor-not-allowed inline-flex items-center w-fit ${
  props.variant === 'block'
    ? blockClasses + 'bg-teal-600'
    : props.variant === 'block-alt'
      ? blockClasses + 'bg-green-600'
      : 'outline-0 focus:ring-2 focus:ring-yellow-400'
} ${props.className ?? ''}`;
```

Reply:
😬 Yeah definitely keep it as you already it. Wonder what I was thinking of then, recall coming across issues todo with dynamic strings and tailwind

Reply:
One thing I like about doing it this way is it's easy for line breaks and spaces from the template string to make it into the class string in the HTML and it can be a bit ugly. That happened with the previous way before this change.